hack/build/docker/builder/Dockerfile

```diff
@@ -1,14 +1,20 @@
-FROM quay.io/centos/centos:stream9
```

Comment:
so we have just switched from fedora to centos stream and now back to fedora @maya-r @mhenriks is that ok?

Reply:
the centos builder had some issues, I hope with https://github.com/kubevirt/project-infra/pull/1824 it will be fixed so we don't need to go back.

Reply:
This change introduces `FROM registry.fedoraproject.org/fedora-minimal:33` instead of centos, 
I remember that we have just switched to centos. If I remember you did the change @maya-r. That is why I have asked for your opinion.

So my question is: What was the reason to change from fedora to centos, and why now we go back to fedora? and which one is better?

Reply:
@maya-r there are additional issues with centos stream.  Specifically the way bazel installs rpms in containers.  So I would like to go back to fedora (although we technically never used the centos builder).  Will make bazel changes in another pr

Reply:
I have been able to update bazel/gazelle/k8s libraries in [this branch](https://github.com/maya-r/containerized-data-importer/tree/wip-k8s-1.23), but I don't mind going back to fedora and re-doing the centos builder change if it is easier for you that way.
(the centos stream CI connectivity issue should be resolved now)

Reply:
Great, let's get this massive pr out of the way then integrate your builder/bazel changes